grpc/src/main/java/io/grpc/netty/shaded/io/grpc/netty/CustomChannelFactory.java

```diff
@@ -0,0 +1,83 @@
+package io.grpc.netty.shaded.io.grpc.netty;
+
+import io.grpc.internal.SharedResourcePool;
+import io.grpc.netty.shaded.io.netty.channel.Channel;
+import io.grpc.netty.shaded.io.netty.channel.ChannelFactory;
+import io.grpc.netty.shaded.io.netty.channel.EventLoopGroup;
+import io.grpc.netty.shaded.io.netty.channel.ServerChannel;
+import io.grpc.netty.shaded.io.netty.channel.group.ChannelGroup;
+import io.grpc.netty.shaded.io.netty.channel.group.DefaultChannelGroup;
+import io.grpc.netty.shaded.io.netty.util.concurrent.GlobalEventExecutor;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tracks all the created sessions. It wraps the {@link Utils#DEFAULT_SERVER_CHANNEL_FACTORY} and
+ * uses it for session creation. Once it's notified via the {@code closeFilter()} method, it closes
+ * all channels matching the predicate.
+ */
+public class CustomChannelFactory implements ChannelFactory<ServerChannel> {
+  private static final Logger logger = LoggerFactory.getLogger(CustomChannelFactory.class);
+  private final ChannelFactory<? extends ServerChannel> channelFactory;
+  private final List<Channel> channels = new ArrayList<>();
+
+  public CustomChannelFactory() {
+    this.channelFactory = Utils.DEFAULT_SERVER_CHANNEL_FACTORY;
```

Comment:
pass to the `CustomChannelFactory(ChannelFactory<? extends ServerChannel> channelFactory)`, this is the only way to have this unit tested and pass mocked `ChannelFactory`